test/localDate.test.ts

```diff
@@ -507,6 +507,206 @@ describe('A value object representing a local date', () => {
+            period: Period.ofYears(2),
+            expected: LocalDate.of(2015, 8, 31),
+        },
+        '2015-08-31 - P1Y2M3D': {
+            start: LocalDate.of(2016, 11, 3),
+            period: Period.of(1, 2, 3),
+            expected: LocalDate.of(2015, 8, 31),
+        },
+        '2016-11-03 - P-1Y-2M-': {
+            start: LocalDate.of(2015, 8, 31),
+            period: Period.of(-1, -2, -3),
+            expected: LocalDate.of(2016, 11, 3),
+        },
+    }))('should subtract a period from a local date %s', (_, scenario) => {
+        const result = scenario.start.minus(scenario.period);
+
+        expect(result.toString()).toStrictEqual(scenario.expected.toString());
+    });
+
+    type PeriodScenario = {
+        start: LocalDate,
+        end: LocalDate,
+        expected: Period,
+    };
+
+    it.each(Object.entries<PeriodScenario>({
+        '2015-08-31 / 2015-08-31': {
+            start: LocalDate.of(2015, 8, 31),
+            end: LocalDate.of(2015, 8, 31),
+            expected: Period.zero(),
+        },
+        '2015-08-31 / 2015-09-01': {
+            start: LocalDate.of(2015, 8, 31),
+            end: LocalDate.of(2015, 9, 1),
+            expected: Period.ofDays(1),
+        },
+        '2015-08-31 / 2015-09-02': {
+            start: LocalDate.of(2015, 8, 31),
+            end: LocalDate.of(2015, 9, 2),
+            expected: Period.ofDays(2),
+        },
+        '2015-08-31 / 2015-09-07': {
+            start: LocalDate.of(2015, 8, 31),
+            end: LocalDate.of(2015, 9, 7),
+            expected: Period.ofWeeks(1),
+        },
+        '2015-08-31 / 2015-09-14': {
+            start: LocalDate.of(2015, 8, 31),
+            end: LocalDate.of(2015, 9, 14),
+            expected: Period.ofWeeks(2),
+        },
+        '2015-08-31 / 2015-10-01': {
+            start: LocalDate.of(2015, 8, 31),
+            end: LocalDate.of(2015, 10, 1),
+            expected: Period.of(0, 1, 1),
+        },
+        '2015-08-31 / 2015-10-31': {
+            start: LocalDate.of(2015, 8, 31),
+            end: LocalDate.of(2015, 10, 31),
+            expected: Period.ofMonths(2),
+        },
+        '2015-08-31 / 2016-08-31': {
+            start: LocalDate.of(2015, 8, 31),
+            end: LocalDate.of(2016, 8, 31),
+            expected: Period.ofYears(1),
+        },
+        '2015-08-31 / 2017-08-31': {
+            start: LocalDate.of(2015, 8, 31),
+            end: LocalDate.of(2017, 8, 31),
+            expected: Period.ofYears(2),
+        },
+        '2015-08-31 / 2016-11-04': {
+            start: LocalDate.of(2015, 8, 31),
+            end: LocalDate.of(2016, 11, 3),
+            expected: Period.of(1, 2, 3),
+        },
+        '2016-11-04 / 2015-08-31': {
+            start: LocalDate.of(2016, 11, 3),
+            end: LocalDate.of(2015, 8, 31),
+            expected: Period.of(-1, -2, -3),
+        },
+    }))('should calculate the period between local dates %s', (_, scenario) => {
```

Comment:
```suggestion
    }))('should calculate the period between $start and $end', (_, scenario) => {
```

The keys are not needed